x-pack/plugins/security_solution/public/dashboards/components/dashboard_renderer.tsx

```diff
@@ -37,23 +43,47 @@ const DashboardRendererComponent = ({
     to: string;
     toStr?: string | undefined;
   };
+  viewMode?: ViewMode;
 }) => {
+  const { embeddable } = useKibana().services;
   const dispatch = useDispatch();
-  const [dashboardContainer, setDashboardContainer] = useState<DashboardAPI>();
 
-  const getCreationOptions = useCallback(
+  const securityTags = useSecurityTags();
+  const firstSecurityTagId = securityTags?.[0]?.id;
+
+  const isCreateDashboard = !savedObjectId;
+
+  const getCreationOptions: () => Promise<DashboardCreationOptions> = useCallback(
     () =>
       Promise.resolve({
-        getInitialInput: () => ({ timeRange, viewMode: ViewMode.VIEW, query, filters }),
+        useSessionStorageIntegration: true,
         useControlGroupIntegration: true,
+        getInitialInput: () => ({
+          timeRange,
+          viewMode,
+          query,
+          filters,
+        }),
+        getIncomingEmbeddable: () =>
+          embeddable.getStateTransfer().getIncomingEmbeddablePackage(APP_UI_ID, true),
+        getEmbeddableContainerContext: (dashboardId?: string) => ({
+          getCurrentPath: () =>
+            dashboardId ? `dashboards/${dashboardId}/edit` : `dashboards/create`,
```

Comment:
It would be great if we avoided using hardcoded paths, if possible.
Can we use the constant?
```suggestion
            dashboardId ? `${DASHBOARDS_PATH}/${dashboardId}/edit` : `${DASHBOARDS_PATH}/create`,
```

or get the current `pathname` from `useLocation()`?